third_party/terraform/resources/resource_dataflow_flex_template_job.go.erb

```diff
@@ -0,0 +1,154 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	dataflow "google.golang.org/api/dataflow/v1b3"
+)
+
+// NOTE: resource_dataflow_flex_template currently does not support updating existing jobs.
+// Changing any non-computed field will result in the job being deleted (according to its
+// on_delete policy) and recreated with the updated parameters.
+
+// resourceDataflowFlexTemplateJob defines the schema for Dataflow FlexTemplate jobs.
+func resourceDataflowFlexTemplateJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataflowFlexTemplateJobCreate,
+		Read:   resourceDataflowFlexTemplateJobRead,
+		// Update: resourceDataflowFlexTemplateJobUpdateByReplacement,
+		// Note: resourceDataflowJobDelete is borrowed from dataflow_job to maintain
+		// same cancel/drain on_delete behavior.
+		Delete: resourceDataflowJobDelete,
+		Schema: map[string]*schema.Schema{
+
+			"container_spec_gcs_path": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"on_delete": {
```

Comment:
I _believe_ we want a blank update method defined. That will make Terraform persist the value for `on_delete` into state. If there's no method at all, it requires this field to recreate the resource.